src/logic/hooks/useExplorerInfo.tsx

```diff
@@ -0,0 +1,30 @@
+import { useSelector } from 'react-redux'
+import { providerSelector } from '../wallets/store/selectors'
+import { getNetworkConfig } from 'src/config/utils'
+
+export type BlockScanInfo = () => {
+  alt?: string
+  url?: string
+}
+
+export const useExplorerInfo = (hash?: string | null): BlockScanInfo => {
+  const provider = useSelector(providerSelector)
+  const config = getNetworkConfig(provider.network)
+
+  const blockScanInfo = () => {
+    const result = {
+      url: '',
+      alt: '',
+    }
+    if (!hash) {
+      return result
+    }
+    const type = hash.length > 42 ? 'tx' : 'address'
+    result.url = `${config?.environment.dev?.networkExplorerUrl}${type}/${hash}`
+    result.alt = config?.environment.dev?.networkExplorerName || ''
```

Comment:
I know we discussed this in a meeting.

But given that the environment/network will be defined at build time, reacting to the `provider` change doesn't seem to be necessary.

Furthermore, we refactored the `src/config/index.ts` service, and we expose the current network's explorer info (https://github.com/gnosis/safe-react/pull/1422/files#diff-f7ebe2d5e45a18ff255acd348152301aR71-R75).

If the later is not done here, we must consider using `getNetworkExplorerInfo`, and all the getters that that service provides, before finishing the xDai feature.